packages/base-crypto/src/VirgilCrypto.ts

```diff
@@ -457,16 +687,24 @@ export class VirgilCrypto implements ICrypto {
       }
     }
 
-    const signature = messageInfoCustomParams.findData(DATA_SIGNATURE_KEY);
+    let signature: Uint8Array | undefined;
+    try {
+      signature = messageInfoCustomParams.findData(DATA_SIGNATURE_KEY);
+    } catch (error) {
+      lowLevelPrivateKey.delete();
+      recipientCipher.delete();
+      messageInfoCustomParams.delete();
+      throw error;
+    }
 
-    const isValid = this.verifySignature(decryptedData, signature, signerPublicKey);
+    const isValid = this.verifySignature(decryptedData, signature!, signerPublicKey);
     if (!isValid) {
       throw new Error('Invalid signature');
```

Comment:
See comment in `decryptThenVerify` about releasing memory before throwing custom errors.